session/src/main/java/org/apache/iotdb/session/Session.java

```diff
@@ -1257,13 +1283,15 @@ private void insertTabletsWithLeaderCache(Map<String, Tablet> tablets, boolean s
     EndPoint endPoint;
     SessionConnection connection;
     Map<SessionConnection, TSInsertTabletsReq> tabletGroup = new HashMap<>();
+    Map<SessionConnection, EndPoint> sessionConnectionEndPointMap = new HashMap<>();
     for (Entry<String, Tablet> entry : tablets.entrySet()) {
       endPoint = deviceIdToEndpoint.get(entry.getKey());
       if (endPoint != null) {
         connection = endPointToSessionConnection.get(endPoint);
       } else {
         connection = defaultSessionConnection;
       }
+      sessionConnectionEndPointMap.putIfAbsent(connection, endPoint);
```

Comment:
As `IoTDBConnectionException` is rarely thrown, I doubt whether this is worth recording even without errors, maybe we can traverse `endPointToSessionConnection` to delete expired sessionConnection when `IoTDBConnectionException` is thrown.

Reply:
That's another solution. the original though is as the number of connections is raly small, so I think it ok add one map here and trade space for time.
As you say, maybe `IoTDBConnectionException` is rarely thrown, so I agree with you.